### DISCUSSION TITLE
Using GPU on Mac M1?

### DISCUSSION THREAD

**Discussion by KaiD1993** [hacksider/Deep-Live-Cam#403] [Q&A]
Any way to use GPU on Mac M1 ?

**Comment by UltraSeeder** [comment]
python3 run.py --execution-provider coreml

**Reply by KaiD1993** [reply]
still superslow

**Comment by Asthar0th** [comment]
it's very superslow basically useless

**Reply by hacksider** [reply]
> it's very superslow basically useless

Thanks for the feedback, you can actually use alternative tools if needed. 😅

**Comment by ghost** [comment]
Mac is not made for AI

**Comment by BG-Hanna** [comment]
hey there @KaiD1993,
I personally have a M2 mac air, and from my experience running AI heavy tasks locally isn't ideal. CoreML can help, but as you said it’s still much slower than dedicated GPUs.
**Solution**
I would recommend using a cloud GPU (Google Colab, AWS, Paperspace).

**Comment by james76896** [comment]
C:\Users\XPS\Deep-Live-cam>python run.py --execution-provider tensorrt
Frame processor face_enhancer not found
Frame processor face_enhancer not found

C:\Users\XPS\Deep-Live-cam>python run.py --execution-provider cpu
Frame processor face_enhancer not found
Frame processor face_enhancer not found

i have been having this 
what do i do please
